src/reanimated2/hook/useAnimatedSensor.ts

```diff
@@ -138,5 +138,5 @@ export function useAnimatedSensor(
     };
   }, [sensorType, userConfig]);
 
-  return ref.current;
+  return ref.current as AnimatedSensor<ValueRotation> | AnimatedSensor<Value3D>;
```

Comment:
the return type here depends on `SensorType`, can we enforce it, perhaps with function overloading?

Reply:
It is enforced in function overloading starting at line 66. The problem is that Typescript doesn't care about function declaration types in function implementation typing, and it requires it to handle union of input types and union of return types in EVERY configuration possible.